execution/group_by.go

```diff
@@ -20,9 +21,10 @@ type Aggregate interface {
 }
 
 type GroupBy struct {
-	storage storage.Storage
-	source  Node
-	key     []Expression
+	storage   storage.Storage
+	source    Node
+	watermark Node
```

Comment:
GroupBy shouldn't have a watermark source.

The PullEngine above it does. Please look at how an execution group by is created from a physical groupBy.

The pull engine will pull watermarks from the watermark source, and update the watermark in the processbykey node. Which will trigger the necessary keys afterwards.

Reply:
No changes to execution/group_by.go should be necessary. Only to how the physical plan injects the watermark source into the pull engine.